src/command/subgraph/publish.rs

```diff
@@ -74,4 +112,183 @@ impl Publish {
             publish_response,
         })
     }
+
+    fn handle_maybe_invalid_routing_url(
```

Comment:
let's convert this from an associated function to a method on the struct
```suggestion
    fn handle_maybe_invalid_routing_url(&self
```

then we can use `&self.routing_url` instead of needing to pass `self.routing_url.clone()`

Reply:
Discussed offline - having this fn not require `self` is great for testing purposes (so we're now using a reference to the `routing_url` instead of cloning it).
35d6dc4